lib/seam/options.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module SeamOptions
+  def self.get_endpoint(endpoint = nil)
+    endpoint || get_endpoint_from_env || Seam::DEFAULT_ENDPOINT
+  end
+
+  def self.get_endpoint_from_env
+    seam_api_url = ENV["SEAM_API_URL"]
+    seam_endpoint = ENV["SEAM_ENDPOINT"]
+
+    if seam_api_url
+      warn "\033[93mUsing the SEAM_API_URL environment variable is deprecated. Support will be removed in a later major version. Use SEAM_ENDPOINT instead.\033[0m"
+    end
+
+    if seam_api_url && seam_endpoint
+      warn "\033[93mDetected both the SEAM_API_URL and SEAM_ENDPOINT environment variables. Using SEAM_ENDPOINT.\033[0m"
+    end
+
+    seam_endpoint || seam_api_url
+  end
+
+  class SeamHttpInvalidOptionsError < StandardError
+    def initialize(message)
+      super("SeamHttp received invalid options: #{message}")
+    end
+  end
+
+  def self.seam_http_options_with_api_key?(api_key: nil, personal_access_token: nil)
+    return false if api_key.nil?
+
+    if personal_access_token
+      raise SeamHttpInvalidOptionsError.new(
+        "The personal_access_token option cannot be used with the api_key option"
+      )
+    end
+
+    true
+  end
```

Comment:
Just realized we made this naming mistake on the python repo as well (we should fix). Because of how the JS packages and modules are broken up, SeamHttp as a class only exists in the javascript sdk (it's aliased to Seam). All of the other SDKs only have Seam. So any error or name that refers to to Seam should be Seam*, not SeamHttp*.

```suggestion
  def self.seam_options_with_api_key?(api_key: nil, personal_access_token: nil)
    return false if api_key.nil?

    if personal_access_token
      raise SeamInvalidOptionsError.new(
        "The personal_access_token option cannot be used with the api_key option"
      )
    end

    true
  end
```

But `SeamHttpApiError` should stay as `SeamHttpApiError` since this is a Seam Error from a response from the HTTP API. Basically, the Javascript SDK happens to have this slight ambiguity, but all the other SDKs don't.

Reply:
Oh nice catch! Sure, I agree. Will fix it on python as well.